sretoolbox/utils/k8s.py

```diff
@@ -0,0 +1,35 @@
+import re
+import uuid
+
+
+def unique_job_name(
+    name: str, prefix: str = "", suffix: str = "", max_length: int = 63
+) -> str:
+    """Generate a unique Kubernetes job name with an optional given prefix.
+
+    Args:
+        name (str): Base name for the job.
+        prefix (str, optional): Prefix to add to the job name. Defaults to "".
+        suffix (str, optional): Suffix to add to the job name. Defaults to a random UUID.
+
+    Returns:
+        str: A unique job name.
+    """
```

Comment:
Do we need to support prefix? There is an edge case that if provided prefix length is greater than max_length, then the result is not a valid job name. Same case for suffix.

We also need to doc that if suffix is provided, the unique is by caller, the return is just a valid job name. And default suffix is 8 random chars, join name with `-`.

If we just handle name and suffix, error handling is easy:

```python
def unique_job_name(
    name: str, suffix: str = "", max_length: int = 63
) -> str:
    if len(suffix) >= max_length:
        raise XXX
    if not suffix:
        suffix = uuid.uuid4().hex[:8]
    if not suffix.startswith("-"):
        suffix = "-" + suffix
    name = name[: max_length - len(suffix)]
    ...
```

Reply:
Dropped the prefix in the last commit